src/Orleans.Streaming/Internal/StreamConsumerExtension.cs

```diff
@@ -187,17 +187,17 @@ public Task<StreamHandshakeToken> GetSequenceToken(GuidId subscriptionId)
 
         internal int DiagCountStreamObservers<T>(InternalStreamId streamId)
         {
-            return allStreamObservers.Values
-                                     .OfType<StreamSubscriptionHandleImpl<T>>()
-                                     .Aggregate(0, (count, o) => count + (o.SameStreamId(streamId) ? 1 : 0));
+            return allStreamObservers.Count(o => o.Value is StreamSubscriptionHandleImpl<T> i && i.SameStreamId(streamId));
         }
 
-        internal IList<StreamSubscriptionHandleImpl<T>> GetAllStreamHandles<T>()
+        internal List<StreamSubscriptionHandleImpl<T>> GetAllStreamHandles<T>()
         {
-            return allStreamObservers.Values
-                .OfType<StreamSubscriptionHandleImpl<T>>()
-                .Where(o => o != null)
-                .ToList();
+            var ls = new List<StreamSubscriptionHandleImpl<T>>();
+            foreach (var o in allStreamObservers)
+            {
+                if (o.Value is StreamSubscriptionHandleImpl<T> i) ls.Add(i);
```

Comment:
We should use the `is {} i` pattern to make ourselves immune to refactoring. Eg, if allStreamObservers' element type changes, then this condition may always be false.

Reply:
My mistake - this is a downcast